client_connection/manager_test.go

```diff
@@ -0,0 +1,141 @@
+package client_connection
+
+import (
+	"errors"
+	"github.com/mysterium/node/communication"
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/openvpn"
+	"github.com/mysterium/node/openvpn/service_discovery"
+	"github.com/mysterium/node/openvpn/session"
+	"github.com/mysterium/node/server"
+	mysterium_api_client "github.com/mysterium/node/server/dto"
+	"github.com/mysterium/node/service_discovery/dto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"sync"
+	"testing"
+)
+
+type test_context struct {
+	suite.Suite
+	connManager         *connectionManager
+	fakeDiscoveryClient server.Client
```

Comment:
Specify more exact type, if You know that it is fake

Reply:
Good point

Reply:
Actually I cannot do that, as server.NewClientFake() returns Client interface already.

Reply:
Decided: NewClientFake() should return not interface but concrete structure.